### PR TITLE
Bug 1170089 - [Settings] Refactor simcard_fdn.js, r=Arthur

### PR DESCRIPTION
**Pull request by EragonJ**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b9a4a79b367d7be888ba42e2611303b15ebd5ef6

File: apps/settings/js/panels/call_fdn_settings/panel.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
+define(function(require) {
+  'use strict';
+
+  var SettingsPanel = require('modules/settings_panel');
+  var DialogService = require('modules/dialog_service');
+
+  return function ctor_call_fdn_settings_panel() {
+    return SettingsPanel({
+      onInit: function(panel, options) {
+        this._cardIndex = options.cardIndex || 0;
```
Comment (crh0716) [inline apps/settings/js/panels/call_fdn_settings/panel.js]: When back from the fdn list, there won't be `cardIndex` but we still need to use the same card index, so we need to cache the index.

File: apps/settings/js/panels/call_fdn_settings/panel.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        this._conns = window.navigator.mozMobileConnections;
+        this._conn = this._conns[this._cardIndex];
+
+        this._elements = {};
```
Comment (crh0716) [inline apps/settings/js/panels/call_fdn_settings/panel.js]: nit: can we initial the object with the elements?

File: apps/settings/test/unit/panels/call_fdn_settings/panel_test.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        callFdnSettingsPanel.onInit(document.body, {});
+        done();
+    });
+  });
+
+  teardown(function() {
+    document.body.innerHTML = '';
+  });
+
+  suite('onShow >', function() {
```
Comment (crh0716) [inline apps/settings/test/unit/panels/call_fdn_settings/panel_test.js]: nit: `onBeforeShow`.

File: apps/settings/js/panels/call_fdn_settings/panel.js (hunk 2 above)
Comment (EragonJ) [inline apps/settings/js/panels/call_fdn_settings/panel.js]: done

File: apps/settings/test/unit/panels/call_fdn_settings/panel_test.js (hunk 3 above)
Comment (EragonJ) [inline apps/settings/test/unit/panels/call_fdn_settings/panel_test.js]: done

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1b470764b2cec6d108bc3ca1005f75065fae5c25

**Pull request merged**